src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -0,0 +1,295 @@
+        // TODO: Consider multi-threading?
+        Map<ZonedDateTime, Itinerary> datesWithMatchingItineraries = new HashMap<>();
+
+        Collection<Map.Entry<ZonedDateTime, String>> entriesToIterate = sortDates
+            ? queriesByDate.entrySet().stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toList())
+            : queriesByDate.entrySet();
+
+        for (Map.Entry<ZonedDateTime, String> entry : entriesToIterate) {
+            ZonedDateTime date = entry.getKey();
+            String params = entry.getValue();
+
+            OtpDispatcherResponse response = OtpDispatcher.sendOtpPlanRequest(params);
+            TripPlan plan = response.getResponse().plan;
+            if (plan != null && plan.itineraries != null) {
+                for (Itinerary itinerary: plan.itineraries) {
+                    // If a matching itinerary is found, save the date with the matching itinerary.
+                    // The matching itinerary will replace the original trip.itinerary.
+                    if (itinerariesMatch(trip.itinerary, itinerary)) {
+                        datesWithMatchingItineraries.put(date, itinerary);
+                    }
+                }
+            }
+        }
+        return new ItineraryExistence(queriesByDate.keySet(), datesWithMatchingItineraries);
+    }
+
+    /**
+     * Overload used in normal conditions.
+     */
+    public static ItineraryExistence checkItineraryExistence(MonitoredTrip trip, boolean checkAllDays) throws URISyntaxException {
+        return checkItineraryExistence(trip, checkAllDays,false);
+    }
+
+    /**
+     * TODO: fully implement
+     */
+    public static boolean itineraryIsSavable(Itinerary itinerary) {
+        return true;
+    }
+
+    /**
+     * Returns true if the itineraries match for the purposes of trip monitoring.
+     *
+     * @param previousItinerary The original itinerary that others are compared against.
+     * @param newItinerary A new itinerary that might match the previous itinerary.
+     */
+    public static boolean itinerariesMatch(Itinerary previousItinerary, Itinerary newItinerary) {
+        // make sure either itinerary is savable before continuing
+        if (!itineraryIsSavable(previousItinerary) || !itineraryIsSavable(newItinerary)) return false;
+
+        // make sure itineraries have same amount of legs
+        if (previousItinerary.legs.size() != newItinerary.legs.size()) return false;
+
+        // make sure each leg matches
+        for (int i = 0; i < previousItinerary.legs.size(); i++) {
+            Leg previousItineraryLeg = previousItinerary.legs.get(i);
+            Leg newItineraryLeg = newItinerary.legs.get(i);
+
+            // for now don't analyze non-transit legs
+            if (!previousItineraryLeg.transitLeg) continue;
+
+            // make sure the same from/to stop are being used
```

Comment:
Similar to `itinerariesMacth` and `stopsMatch`, there should be a `legsMatch` function that picks up code from here.

Reply:
Fixed in 52d438aad1badb194c212a736a16d72aff3da526